src/reactapp/src/context/Form/ShippingAddress/ShippingAddressFormManager.jsx

```diff
@@ -47,98 +47,289 @@ const validationSchema = {
+  const shippingCountry = _get(values, shippingAddrCountryField);
+  const stateInfo = _get(stateList, `${shippingCountry}`, []).find(
+    s => s.code === shippingRegion
+  );
 
   const formSubmit = useCallback(
-    async formValues => {
-      setPageLoader(true);
+    async (formValues, customerAddressId) => {
+      try {
+        setPageLoader(true);
 
-      await addCartShippingAddress(shippingAddrFieldValues);
+        const shippingAddressToSave = _get(formValues, SHIPPING_ADDR_FORM);
+        let updateShippingAddress = _makePromise(
+          addCartShippingAddress,
+          shippingAddressToSave
+        );
+        let updateBillingAddress = () => {};
 
-      const isBillingSame = _get(
-        formValues,
-        'billing_address.isSameAsShipping'
-      );
+        if (customerAddressId) {
+          updateShippingAddress = _makePromise(
+            setCustomerAddressAsShippingAddress,
+            customerAddressId
+          );
+        }
 
-      if (isBillingSame) {
-        await setCartBillingAddress({
-          ...shippingAddrFieldValues,
-          isSameAsShipping: true,
-        });
-      }
+        const isBillingSame = _get(formValues, isSameAsShippingField);
 
-      setFormEditMode(false);
-      setPageLoader(false);
+        if (isBillingSame) {
+          if (customerAddressId) {
+            updateBillingAddress = _makePromise(
+              setCustomerAddressAsBillingAddress,
+              customerAddressId,
+              isBillingSame
+            );
+          } else {
+            updateBillingAddress = _makePromise(setCartBillingAddress, {
+              ...shippingAddressToSave,
+              isSameAsShipping: true,
+            });
+          }
+        }
+
+        await Promise.all([updateShippingAddress(), updateBillingAddress()]);
+
+        setFormEditMode(false);
+        setPageLoader(false);
+      } catch (error) {
+        console.log({ error });
+        setPageLoader(false);
+      }
     },
     [
-      shippingAddrFieldValues,
       addCartShippingAddress,
       setCartBillingAddress,
       setFormEditMode,
       setPageLoader,
+      setCustomerAddressAsShippingAddress,
+      setCustomerAddressAsBillingAddress,
     ]
   );
 
-  const {
-    addressId: selectedAddressId,
-    address: selectedShippingAddress,
-  } = useMemo(() => {
-    const addressId = _get(shippingAddressIds, 0);
-    const address = _get(shippingAddressList, addressId);
-    return {
-      addressId,
-      address,
-    };
-  }, [shippingAddressIds, shippingAddressList]);
-
-  // for guest cart, we are setting the first shipping address as the selected
-  // shipping adddress here;
+  const resetShippingAddressFormFields = useCallback(() => {
+    setFieldValue(SHIPPING_ADDR_FORM, { ...shippingAddressFormInitValues });
+  }, [setFieldValue]);
+
+  const saveAddressHandler = useCallback(
+    async formikValues => {
+      try {
+        let promise1 = () => {};
```

Comment:
For consistency, I guess we could use _emptyFunc here again, or do it inline everywhere :)

Reply:
Valid point. I've created `_emptyFunc` helper function later due to its repeated usage. Missed to use in some of the above cases which implemented in prior to that. Changes are done as per the instruction 👍